client/src/main/scala/drt/client/components/MonthlyStaffing.scala

```diff
@@ -41,15 +42,39 @@ object MonthlyStaffing {
   }
 
   def slotsInDay(date: SDateLike, slotDuration: Int): Seq[SDateLike] = {
-    val minutesInDay = 24 * 60
+
     val startOfDay = SDate.midnightOf(date)
-    val slots = minutesInDay / slotDuration
+    val slots = minutesInDay(date) / slotDuration
     List.tabulate(slots)(i => {
       val minsToAdd = i * slotDuration
       startOfDay.addMinutes(minsToAdd)
     })
   }
 
+  def timeZoneSafeTimeSlots(slots: Seq[SDateLike], slotMinutes: Int): Seq[Option[SDateLike]] = {
+    val slotsInRegularDay = (24 * 60 * 60) / slotMinutes
+    val slotsPerHour = 60 / slotMinutes
+    if (slots.size < slotsInRegularDay) {
+
+      val timeslotsWithNones: Seq[Option[SDateLike]] = slots.sliding(2).flatMap(dates =>
+        if (dates(0).getTimeZoneOffsetMillis < dates(1).getTimeZoneOffsetMillis)
```

Comment:
Does sliding() def always give you elements of the size requested? ie does it not give you a single last element if there's an odd number in the original collection?